lib/main/main_model.dart

```diff
@@ -0,0 +1,45 @@
+
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_pk/caches/user.dart';
+import 'package:flutter_pk/global.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+class GoogleSignInApi {
+  Future<String> handleGoogleSignIn() async {
```

Comment:
This seems to be doing more that one thing:
1. Sign in using google
2. Get user from firestore
3. Create user if not exists

User fetching and creating can be part of a different class. The login API logic should not know about user finding and creating.